### ISSUE TITLE
looking for an easy way to create venv on AWX K8s install

### ISSUE THREAD

**Issue by rdzielicki** [ansible/awx#10060]
##### ISSUE TYPE
 - Implement an easy way to manage and create venv in web /task container, by default containers are configured with no privilege escalation

##### SUMMARY
Have an easy way to create, manage venv in AWX in K8s

**Comment by rdzielicki** [comment]
Guess have to build custom images but it so common to need venv and modules that it should be built in ( event if you can wrap easily instances in K8s to rebuilt them),

**Comment by rdzielicki** [comment]
duplicate of #3145 even if the built in feature would be nice ( even if we can rebuilt  and keep persistent data with K8s philosophy)

**Comment by shanemcd** [comment]
- What version of AWX?
- Why do you need to install a custom venv in the web container?

**Comment by rdzielicki** [comment]
tried AWX version 15 in K8s configuration, but update it to latest soon. Got to correct an information i need it in awx_task, so you can have a collection of venv with different ansible version and modules; it would be a non sense to pull x AWX pod instances to have several Ansible version and modules.
Also even without the need of venv and different version of Ansible you would need some additionnal modules in Ansible ( forti, winrm, etc etc)

**Comment by daenney** [comment]
You can do something like this to create your own execution environment, with whatever you want in it. Then update your templates to use that execution environment when running the task.

```
FROM quay.io/ansible/awx-ee:0.1.1

USER root

ADD collections.yml /tmp/collections.yml
ADD requirements.txt /tmp/requirements.txt

RUN ansible-galaxy collection install -r /tmp/collections.yml --collections-path /usr/share/ansible/collections
RUN pip install -r /tmp/requirements.txt  # for any Python modules the collections didn't pull in

USER 1000
```

**Comment by rdzielicki** [comment]
> You can do something like this to create your own execution environment, with whatever you want in it. Then update your templates to use that execution environment when running the task.
> 
> ```
> FROM quay.io/ansible/awx-ee:0.1.1
> 
> USER root
> 
> ADD collections.yml /tmp/collections.yml
> ADD requirements.txt /tmp/requirements.txt
> 
> RUN ansible-galaxy collection install -r /tmp/collections.yml --collections-path /usr/share/ansible/collections
> RUN pip install -r /tmp/requirements.txt  # for any Python modules the collections didn't pull in
> 
> USER 1000
> ```

yes but it you'll need to rebuild a DockerFile and pull the image every time you need additionnal Ansible version

**Comment by stanislaspiron** [comment]
@daenney thank you for providing this solution… 

This solution should be documented in awx-ee readme.

**Comment by daenney** [comment]
> > You can do something like this to create your own execution environment, with whatever you want in it. Then update your templates to use that execution environment when running the task.
> > ```
> > FROM quay.io/ansible/awx-ee:0.1.1
> > 
> > USER root
> > 
> > ADD collections.yml /tmp/collections.yml
> > ADD requirements.txt /tmp/requirements.txt
> > 
> > RUN ansible-galaxy collection install -r /tmp/collections.yml --collections-path /usr/share/ansible/collections
> > RUN pip install -r /tmp/requirements.txt  # for any Python modules the collections didn't pull in
> > 
> > USER 1000
> > ```
> 
> yes but it you'll need to rebuild a DockerFile and pull the image every time you need additionnal Ansible version

Sure, but that's kinda how things work in the Kubernetes world. Everything's a container and you shouldn't be modifying them at runtime to install different things into them. Much like you wouldn't have 2 ansible versions in the same venv, you'd have 2 separate execution environments.

**Comment by rdzielicki** [comment]
'i'll discuss this this the team and discuss about our options to do so,
